lib/rules/no-duplicate-selectors/index.js

```diff
@@ -42,6 +42,10 @@ const rule = function(actual, options) {
 		// from other rules that share the same parent and the same source.
 		const selectorContextLookup = nodeContextLookup();
 
+		if (_.get(options, 'disallowInList')) {
+			root.nodes.sort((a, b) => a.selector[0].localeCompare(b.selector[0]));
```

Comment:
This will mutate PostCSS tree. It will change user files.

Also, I think fix should be somewhere around these lines https://github.com/stylelint/stylelint/blob/739a05b8536196a87567da360b13886e32218ea0/lib/rules/no-duplicate-selectors/index.js#L78-L83 Because it's the logic for `disallowInList` option.

Reply:
okay, can you please explain what is the difference between `contextSelectorSet` and `selectorList`? Because i found this kind of issue: if css file contains `input, textarea {} textarea{}` then contextSelectorSet will contain `input,textarea`(i mean `selector` in forEach loop) and selectorList will be `textarea`, thus `includes` returns true. But in our case contextSelectorSet contains `textarea` and selectorList is `input,textarea` so includes returns false. I think i dont get the purpose of these two entities right.

Reply:
Sorry, I missed you question. I believe `contextSelectorSet` has lists of individual selectors per their scope. E. g.

```css
a {}
b {}

@media print {
    a {}
    b {}
}
```

The first two selectors (`a`, `b`) on a top level doesn't know about selectors in other scope (inside `@media`) and vice versa. It's my assumption from reading the code, I did verify that in debugger.

`selectorList` is current rule selectors. E. g.:

```css
a, b {}

c, d {}
```

If you're checking second rule it will be `['c', 'd']`.